spacy/tokens/doc.pyx

```diff
@@ -1924,6 +2042,180 @@ cdef int [:,:] _get_lca_matrix(Doc doc, int start, int end):
     return lca_matrix
 
 
+@cython.boundscheck(False)  # Deactivate bounds checking
+cdef void _set_prefix_lengths(
+    const unsigned char* tok_str,
```

Comment:
See above, if these arrays could just be `vector` or `strings`, then they could be passed as (const) references.

Reply:
See my comment above.